Handle pagination for Domains DataViews

merkushin opened the pull request:
Built on top of https://github.com/Automattic/wp-calypso/pull/97915

## Proposed Changes

This implementation relies on solutions from Sites DataViews.

* Store pagination in query params.
* Apply pagination from query params.

## Why are these changes being made?

* We want to store the pagination state so that use can save the link, follow it later and appear in the place they expect.
* We want to preserve pagination params when open the side panel with detailed information.

## Testing Instructions

* Turn on feature flags in your browser's console: window.sessionStorage.setItem('flags', ['calypso/all-domain-management', 'calypso/domains-dataviews']).
* Go to http://calypso.localhost:3000/plugins/manage
* Go to the second page. Make sure pagination params appeared in the URL.
* Change sorting field and dirction, Make sure sorting params appeared in the URL.
* Click on a domain and check if it opened a pane on the right. 
* Make sure pagination and sorting params are still in the URL.
* Make sure the list of domains on the left respects the pagination and sorting params from the URL.
* In the same window (because the feature flags are session-based) open http://calypso.localhost:3000/plugins/manage?page=2.
* Make sure you see the second page of results.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

merkushin committed:
Use domains DataViews in BulkAllDomains with a feature flag

merkushin committed:
Use original backend to load data

merkushin committed:
Store pagination in query params and apply it

merkushin committed:
Revert unnecessary changes

merkushin committed:
Store sorting params in query parameters

m1r0 reviewed (commented):
Hey Dmitry! Overall the pagination works as described, but I've noticed a few quirks:

* When you go to the second page, refresh the page, and use the pagination to go back to page one, it's not working on the first click. The second click works.

https://github.com/user-attachments/assets/8bbdd1f4-cae4-4641-9e5d-94c3dd9284c5

* When you sort and refresh the page, the sort param disappears.

https://github.com/user-attachments/assets/5b3313f8-387a-4cc1-a998-4f617640d9bf

merkushin committed:
Fix sort direction after refresh

merkushin committed:
Fix pagination not working for the first time

merkushin committed:
Set default per page to 50, same as for sites

merkushin commented:
@m1r0 Thanks for catching those bugs.

It took some time to find a typo 😅 Initially was thinking it was one of DataViews' “glitches” https://github.com/Automattic/wp-calypso/pull/97953/commits/9394811208b4f5f8969ad7cb37865cc905b09ce1

Pagination started working properly when I deleted code I took from the Sites' implementation, its purpose was to address another DataViews issue. Though the PR from the comment is still open, I couldn't reproduce the issue now. https://github.com/Automattic/wp-calypso/pull/97953/commits/c7182a7092176e849a82a5d223cd3d1796d19080

Also, I changed the default value for “per page” to 50. The same value is used in Sites.

m1r0 reviewed (commented):
I can confirm that the above issues are fixed, thanks!

I've noticed two more issues:

* When you sort by the `SSL` column and refresh the page, an error is thrown.

![image](https://github.com/user-attachments/assets/d9e522be-a0ff-45f0-a490-23714df0f957)

* A minor UI bug (possibly not related to this PR) is that the sort selection is not reset when you sort by another column.

https://github.com/user-attachments/assets/e3aa3fbd-8993-4527-ab22-b1c02ee5fa06

If you're in a hurry to merge, feel free to do so if the first issue is fixed. The second one is not a blocker.

merkushin committed:
Display loader for cells with lazy loading (#98456)

* Bulk actions notification improvements (#98457)
* Move bulk action notice outside the DataViews container
* Show singular form of the message if there was an update for one domain

merkushin committed:
Return an empty string when domain is not loaded yet

merkushin commented:
Thanks for the catches, @m1r0!

Fixed the first issue here (overlooked it somehow): https://github.com/Automattic/wp-calypso/pull/97953/commits/ec97f281d996bea845cf6fb2533304a5b7b4a4fe 

As for the second one, I found the same behavior in Sites. Probably, it comes from DataViews.
I'll merge the PR to the main one and try to investigate it a bit later.
